File: eidocell-backend/schemas/export.py
from pydantic import BaseModel


class ExportRequest(BaseModel):
    output_directory: str
    include_classes: bool = True
    include_clusters: bool = True
    include_masks: bool = True
    include_binary_masks: bool = False
    include_csv: bool = True


class ExportResult(BaseModel):
    output_directory: str
    classes_exported: int = 0
    clusters_exported: int = 0
    masks_exported: int = 0
    binary_masks_exported: int = 0
    csv_rows: int = 0

File: eidocell-ui/src/types/export.ts
export interface ExportRequest {
  output_directory: string
  include_classes?: boolean
  include_clusters?: boolean
  include_masks?: boolean
  include_binary_masks?: boolean
  include_csv?: boolean
}

export interface ExportResult {
  output_directory: string
  classes_exported: number
  clusters_exported: number
  masks_exported: number
  binary_masks_exported: number
  csv_rows: number
}
